api/scpca_portal/test/models/test_job.py

```diff
@@ -18,6 +18,40 @@ def setUp(self):
             f"{self.mock_project_id}-{self.mock_download_config_name}"
         )
         self.mock_dataset = DatasetFactory()
+        self.aws_batch_job_status = [  # AWS Batch job statuses
+            "SUBMITTED",
+            "PENDING",
+            "RUNNABLE",
+            "STARTING",
+            "RUNNING",
+            "SUCCEEDED",
+            "FAILED",
+        ]
+
+    def bulk_create_mock_jobs(self, list_of_jobs):
+        """Helper to create job instances using JobFactory."""
+        for job in list_of_jobs:
+            JobFactory(
+                batch_job_name=self.mock_project_batch_job_name,
+                batch_job_id=job["batch_job_id"],
+                state=job["state"],
+            )
+
+    def bulk_create_mock_aws_batch_jobs(self, list_of_jobs, terminated_job=None):
+        """Helper to generate AWS Batch jobs for mocked method return value."""
+
+        if terminated_job:
+            list_of_jobs.append(terminated_job)
+
+        return [
+            {
+                "jobId": job["jobId"],
+                "status": job["status"],
+                "statusReason": f"Job {job['status']}",
+                "isTerminated": terminated_job and job["jobId"] == terminated_job["jobId"],
+            }
+            for job in list_of_jobs
+        ]
```

Comment:
I've added these helpers within `TestJob`. Should we move them to `JobFactory`?

Reply:
I think a JobFactory would be an improvement. It would end up being less logic based than the `create_mock...` implementations which we should avoid when writing tests.